api/auth_middleware_test.go

```diff
@@ -8,24 +8,48 @@ import (
 	helpers "github.com/topfreegames/Will.IAM/testing"
 )
 
-func TestAuthMiddlewareKeyPair(t *testing.T) {
+func TestAuthMiddlewareKeyPairShouldAuthenticateUser(t *testing.T) {
 	helpers.CleanupPG(t)
 
-	rootSA := helpers.CreateRootServiceAccount(t)
-
+	rootSA := helpers.CreateRootServiceAccountWithKeyPair(t)
 	app := helpers.GetApp(t)
-
 	req, _ := http.NewRequest(http.MethodGet, "/service_accounts", nil)
+
 	req.Header.Set("Authorization", fmt.Sprintf(
-		"KeyPair %s:%s", rootSA.KeyID, rootSA.KeySecret,
+		"keypair %s:%s", rootSA.KeyID, rootSA.KeySecret,
 	))
-	rec := helpers.DoRequest(t, req, app.GetRouter())
-	if rec.Code != http.StatusOK {
-		t.Errorf("Expected status %d. Got %d", http.StatusOK, rec.Code)
+
+	response := helpers.DoRequest(t, req, app.GetRouter())
+
+	if response.Code != http.StatusOK {
+		t.Errorf("Expected status %d. Got %d", http.StatusOK, response.Code)
 	}
 
-	serviceAccountName := rec.Header().Get("x-service-account-name")
+	serviceAccountName := response.Header().Get("x-service-account-name")
 	if serviceAccountName != "root" {
 		t.Errorf("Expected service account name %s. Got %s", "root", serviceAccountName)
 	}
 }
+
+func TestAuthMiddlewareBearerShouldAuthenticateUser(t *testing.T) {
+	helpers.CleanupPG(t)
+
+	rootSA := helpers.CreateRootServiceAccountWithOAuth(t)
+	tokensRepo := helpers.GetRepo(t).Tokens
+	tokens, _ := tokensRepo.FindByEmail(rootSA.Email)
+	token := tokens[0]
+	app := helpers.GetApp(t)
+	req, _ := http.NewRequest(http.MethodGet, "/service_accounts", nil)
```

Comment:
Don't think so, never is a very strong word 😄 For this test I followed the same pattern as the test for `KeyPair`, which ignored this error... I will handle it and abort the test in case it happens 👍

Reply:
Error handled in ea8d61e9d701777f9fb981b0d3c8dfe10eb1d99a